### ISSUE TITLE
[Warrior] Heroic Strike Miss Part 2

### ISSUE THREAD

**Issue by zenots** [SunwellTracker/issues#1855]
Decription: 

How it works: 

How it should work: 

Source (you should point out proofs of your report, please give us some source):

**Comment by zenots** [comment]
@Riztazz

**Comment by zenots** [comment]
Yeah I guess that 6 minutes was just a fluke?... because I tried again today with a bunch of different setups and all of them had misses, and pretty early too (within 30-60s)

The setups I tried were:

238 hit rating + precision + TG, 2 1 handers
218 hit rating + precision + TG, 2 2 handers
270 hit rating + no talents, 2 1 handers
270 hit rating + precision + no TG, 2 1 handers
263 hit rating + no precision + TG, 2 2 handers

All of those are hit capped one way or another, most a decent bit above hit cap, and also above hit cap from gear AND with precision.

I uploaded the logs of it, but they're kind of rough since at some parts was just me filling up rage bar for testing (mostly before 1handers since I'd run out of rage quickly, so had to test multiple times to see if there was a miss while I still did have rage)
http://www.worldoflogs.com/reports/7mtk1gip14rmj3ec/

I also tested it with Cleave instead of Heroic Strike, with the same results.

**Comment by Riztazz** [comment]
I will look into it as fast as i can :P

**Comment by zenots** [comment]
Found anything that might be causing the problem?

**Comment by Riztazz** [comment]
I will look into it after the weekend, got some stuff to close first ;P
meanwhile @Azathothh could you please do some tests on your own?

**Comment by zenots** [comment]
I just hope it can get fixed before ICC :(

**Comment by zenots** [comment]
With TBC finally being released, I'm hoping there will be a bit of time to sort it out? :D ICC is approaching quickly!

**Comment by zenots** [comment]
bump @Riztazz

**Comment by Kaliram** [comment]
The problem seems to appear when using 2 weapons of the same speed. 
Tested both weapon setups with 3/3 Precision.
![wowscrnshot_121618_213705](https://user-images.githubusercontent.com/24373041/50063740-c2a96f80-017b-11e9-80a2-000e6ddecafb.jpg)
3.4/3.4 speeds, misses.
![wowscrnshot_121618_213929](https://user-images.githubusercontent.com/24373041/50063803-f84e5880-017b-11e9-8613-2a2611c1d3d6.jpg)
3.4/3.6 speed, no misses
Pls fix its massive pain in the ass ty.

**Comment by zenots** [comment]
Should be testing with 3/3 precision and 164 hit rating (5%). But I'm definitely getting misses with 2 of the exact same weapon, so I don't think weapon speed is related, you just have a lot more hit rating than necessary so it's probably giving you pulls with exceptional RNG and no misses.

**Comment by Kaliram** [comment]
That's not what i'm saying. I'm saying that with a 3.4 weapon and 3.6 weapon, i'm getting no misses. With a 3.6 and a 3.4 or a 2 same speed weapons I DO get misses. In other words the only way i'm not getting misses is with a mainhand weapon that is faster than my offhand (screenshot # 2). Not sure how thats not clear, look at the weapons in the screenshots, they are the ones I used each time.

**Comment by zenots** [comment]
You are not over hitcap for off-hand swings with this not working properly :P the hitcap without HS fix is 24%

**Comment by Kaliram** [comment]
There seems to have been a communication breakdown here.

First of all the white attack miss chance while dual-wielding two handed weapons is supposed to be  27%. Source ; http://web.archive.org/web/20130209181158/http://elitistjerks.com/f47/t37804-warrior_fury_compendium/#Understanding_the_Hit_table

Secondly, you seem to be missing the point here. The fix they have implemented is supposed to make offhand strikes that occur while an on next attack ability is queued ignore the dual wield miss chance. Which it IS doing, just only when you are dual wielding with a mainhand that is faster than your offhand, which is not the way this mechanic is supposed to work, it should work with all weapon speeds while dual wielding. And as such, hit rating past 8% should be irrelevant while talking about this fix, and i've no idea why you keep bringing it up.

**Comment by zenots** [comment]
Because you need 5% to reach the cap with 3/3 precision, not 8%. You have 3% extra hit rating which can be skewing your results with an extra layer of RNG.

**Comment by zenots** [comment]
https://gyazo.com/6bdde9b7f7e1fadaadbe5ce1bc286252

1.6 speed main-hand / 3.6 speed offhand, 3 misses. So a faster main-hand does not appear to affect it.

**Comment by Kaliram** [comment]
![wowscrnshot_121718_164956](https://user-images.githubusercontent.com/24373041/50205399-152e8b80-0336-11e9-8ec9-43a97611b763.jpg)
3.4 mainhand / 3.6 offhand.
I'm going to assume that since @Riztazz said in the previous ticket "Basically when you roll out an attack, server checks whether you're dual wielding, if you are then you're added hit chance penalty otherwise you're not, i modified that function to check whether you have on next swing type of spell queued and if so you're not added the DW penalty so you require just base value (8%) to hit." that these offhand swings are simply going off in at a point where the server has not yet registered that a HS/cleave is queued, and 3.4 mh 3.6 oh simply works with this "fix" (maybe 3.4/3.5 too idk. so personal latency and button pressing speed could be factors. All my tests were done with an autohotkey script that hits the button every 10ms, if that helps.

**Comment by zenots** [comment]
Bump @Riztazz
